src/sage/arith/misc.py

```diff
@@ -1970,12 +1970,22 @@ def xgcd(a, b):
 
     INPUT:
 
+    One of the following:
+
     -  ``a, b`` -- integers or more generally, element of a ring for which the
        xgcd make sense (e.g. a field or univariate polynomials).
 
+    -  ``a`` -- a list or tuple of at least two integers or more generally, elements
+       of a ring which the xgcd make sense.
+
     OUTPUT:
 
-    -  ``g, s, t`` -- such that `g = s\cdot a + t\cdot b`
+    One of the following:
+
+    -  ``g, s, t`` -- when two inputs ``a, b`` are given. They satisfy `g = s\cdot a + t\cdot b`.
+
+    -  ``r`` -- a tuple, when only ``a`` is given (and ``b = None``). It has length one longer
+       than the length of ``a``, and its entries satisfy `r_0 = \sum_{i = 0}^{len(a) - 1} r_{i + 1}a_i`.
```

Comment:
So, formally, all the `r_i` could be zero. I think you need to spell out somewhere that `r_0` is the gcd. Maybe at the top of the function.